js/reference.js

```diff
@@ -1018,31 +1154,65 @@ var ERMrest = (function(module) {
-
-                    // replace patterns with values
-                    var colnames = table.columns.names();
-                    for (var c = 0; c < colnames.length; c++) {
-                        var cname = colnames[c];
-                        var search = "{" + cname + "}";
-                        this._displayname = this._displayname.replace(new RegExp(search, 'g'), this._data[cname]);
-                    }
+                var annotation;
+                // If table has table-display annotation then set it in annotation variable
+                if (table.annotations.contains(module._annotations.TABLE_DISPLAY)) {
+                    annotation = module._getRecursiveAnnotationValue(module._contexts.ROWNAME, table.annotations.get(module._annotations.TABLE_DISPLAY).content);
+                }
 
+                // if annotation is populated and annotation has display.rowName property
+                if (annotation && typeof annotation.row_markdown_pattern === 'string') {
+                    var template = annotation.row_markdown_pattern;
+                    
+                    // Get formatted keyValues for a table for the data
+                    var keyValues = module._getFormattedKeyValues(this._pageRef, this._data);
+
+                    // get templated patten after replacing the values using Mustache
+                    var pattern = module._renderTemplate(template, keyValues);
+
+                    // Render markdown content for the pattern
+                    this._displayname = module._formatUtils.printMarkdown(pattern, { inline: true });
+                }
+                // no row_name annotation, use column with title, name, or id:text type
+                else if (typeof this._data.title === 'string') {
+                    col = table.columns.get("title");
+                    this._displayname = col.formatvalue(this._data.title, { context: self._pageRef.context });
+                }
+                else if (typeof this._data.name === 'string') {
+                    col = table.columns.get("name");
+                    this._displayname = col.formatvalue(this._data.name, { context: self._pageRef.context });
+                }
+                else  {  
+
+                    // Check for id column whose type should not be integer or serial
+                    var idCol = table.columns.all().filter(function (c) { 
+                        return ((c.name == "id") && (c.type.name.indexOf('serial') === -1) && (c.type.name.indexOf('int') === -1));  
+                    });
+
+                    // If id column exists
+                    if (idCol.length && typeof this._data.id === 'string') {
+                        this._displayname = idCol[0].formatvalue(this._data.id, { context: self._pageRef.context });
+                    } else {
+                        // Get the columns for shortestKey
+                        var keyColumns = table.shortestKey;
+                        var values = [];
+
```

Comment:
If this consists of all columns, maybe we should return `null` for now. See my comment in the issue.